### ISSUE TITLE
Support autofix

### ISSUE THREAD

**Issue by mitar** [catenacyber/perfsprint#29]
This would be great, so that linter could just autofix the issues.

**Comment by catenacyber** [comment]
It is already there, like perfsprint -h shows you the flag like fix

**Comment by mitar** [comment]
Hm, it is not in golangci-lint:

```
perfsprint: Checks that fmt.Sprintf can be replaced with a faster alternative. [fast: false, auto-fix: false]
```

**Comment by ccoVeille** [comment]
I think you are looking for
- https://github.com/golangci/golangci-lint/issues/1779

**Comment by mitar** [comment]
So the issue is that golangci-lint does not support SuggestedFixes, but some other linters are doing auto-fixing in some other way then?

**Comment by ccoVeille** [comment]
Yes, so it's not an issue that can be addressed in perfsprint

And the ticket already exists in golangci-lint

I'm unsure what is the latest status. I mean I'm unsure if they were not awaiting something to be merged in Go std lib

**Comment by mitar** [comment]
Thanks.

**Comment by catenacyber** [comment]
Thanks Christophe for answering better and faster than me

**Comment by ccoVeille** [comment]
Unless ... I was totally wrong 

https://github.com/golangci/golangci-lint/pull/4994#issuecomment-2338412029

Here is what is recommended apparently https://github.com/4meepo/tagalign/blob/e25313ba1e10f624375cf2573cdcf07473178606/tagalign.go#L163-L173

I think I had also misunderstood and expected golangci-lint to support suggestedfixes, but it's the opposite, it might be to each linter to support the golangci-lint format

**Comment by mitar** [comment]
I mean, or we wait for golangci-lint to support suggestedfixes, or this linter implements custom API golangci-lint requires now.

**Comment by ccoVeille** [comment]
Exactly. But I think it's a simpler move to try supporting the golangci-lint mode maybe

**Comment by catenacyber** [comment]
a PR is welcome if you want golangci-lint mode :-)

**Comment by mitar** [comment]
Let's then reopen this.

**Comment by catenacyber** [comment]
Thanks for watching
